.github/workflows/publish.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+      - "[0-9].[0-9]+.x"
+jobs:
+  checks:
+    name: publication
+    runs-on: ubuntu-latest
+    steps:
+    # we DON'T want to cancel previous runs, especially in the case of a "back to snapshots" build right after a release push
+    - uses: actions/checkout@v2
+    - uses: christian-draeger/read-properties@1.0.1
+      name: read version
+      id: read-version
+      with:
+        path: './gradle.properties'
+        property: 'version'
+    - uses: frabert/replace-string-action@v1.2
+      id: split-qualifier
+      with:
+        string: ${{ steps.read-version.outputs.value }}
+        pattern: '(\d+\.\d+\.\d+)[.-]?([\w-]*)'
```

Comment:
Not sure about the `[.-]?` at the center of the regexp. Isn't one of . or - always present? Why the question mark modifier?

Reply:
old scheme would always have a `.`
new scheme has either a `-` (for anything but releases) or nothing (for releases)

Reply:
Ah, indeed. So it's semantically (.|-?) (which I'm not sure is even valid), and that amounts to your pattern. Gotcha.